src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsoffendercategorisationapi/services/file/S3FileService.kt

```diff
@@ -0,0 +1,105 @@
+package uk.gov.justice.digital.hmpps.hmppsoffendercategorisationapi.services.file
+
+import aws.sdk.kotlin.services.s3.S3Client
+import aws.sdk.kotlin.services.s3.listObjectsV2
+import aws.sdk.kotlin.services.s3.model.DeleteObjectRequest
+import aws.sdk.kotlin.services.s3.model.GetObjectRequest
+import aws.smithy.kotlin.runtime.content.decodeToString
+import aws.smithy.kotlin.runtime.time.toJvmInstant
+import org.apache.commons.lang3.StringUtils
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsoffendercategorisationapi.model.file.FileInfo
+import uk.gov.justice.digital.hmpps.hmppsoffendercategorisationapi.model.file.FileType
+import uk.gov.justice.digital.hmpps.hmppsoffendercategorisationapi.model.file.PendingFile
+import java.io.IOException
+import java.time.ZoneId
+
+@Service
+class S3FileService(
+  private val s3Client: S3Client,
+) {
+
+  suspend fun getLatestViper2File(fileLocation: String, fileType: FileType?): PendingFile? {
+    val bucketAndPrefix = BucketAndPrefix(fileLocation)
+    val s3Result = list(bucketAndPrefix)
+    log.info("Found {} objects in {}", s3Result?.count(), fileLocation)
+
+    val latestViperFile = s3Result
+      ?.filter { o -> o.key?.contains("VIPER_2_") == true }
+      ?.sortedBy { o -> o.lastModified }
+      ?.first()
```

Comment:
This will return the oldest file. I think the intention here is to get the newest file?
```suggestion
      ?.maxByOrNull { o -> o.lastModified }
```